src/pool/srv_pool.c

```diff
@@ -4419,10 +4421,12 @@ pool_extend_map(struct rdb_tx *tx, struct pool_svc *svc,
 	ds_rsvc_request_map_dist(&svc->ps_rsvc);
 
 out_map:
-	if (map_version_p != NULL)
-		*map_version_p = pool_map_get_version((map == NULL || rc != 0) ?
-						      svc->ps_pool->sp_map :
-						      map);
+	if (map_version_p != NULL) {
+		if (map == NULL || rc != 0)
+			*map_version_p = ds_pool_get_version(svc->ps_pool);
+		else
+			*map_version_p = pool_map_get_version(map);
+	}
```

Comment:
The pool_map_get_version calls changed by this PR in this file are actually okay, because they are on xstream 0 and called with svc->ps_lock. ds_pool_get_version is indeed safer though; fine with me.